lib/money-heuristics/analyzer.rb

```diff
@@ -1,21 +1,26 @@
 module MoneyHeuristics
   class Analyzer
     attr_reader :search_tree, :words
-    attr_accessor :str, :currencies
+    attr_accessor :str, :currencies, :methods
 
     def initialize(str, search_tree)
       @str = (str || '').dup
       @search_tree = search_tree
       @currencies = []
+      @methods = { iso_code: "search_by_iso_code", 
+                   symbol: "search_by_symbol", 
+                   name: "search_by_name" }
     end
 
-    def process
+    def process(filters)
       format
       return [] if str.empty?
 
-      search_by_symbol
-      search_by_iso_code
-      search_by_name
+      if (methods.keys & filters).any?
+        filters.each{ |filter| send(methods[filter]) }
+      else
+        methods.values.each{|method| send(method)}
+      end
```

Comment:
To make this a bit more straight-forward, can you first determine the methods and then call them in a single run? Btw, there's a `values_at` method on a Hash, which might come in handy

Reply:
I've tried a bunch of approaches, but didn't manage to make the whole logic fit into one neat line. So I've split a bit more clean logic into a private method to keep only one line `process`. Hope that's good enough, let me know if you have an idea how to refactor further